### PR TITLE
[IMP] core: sanitize request cookies

### PR DESCRIPTION
**Pull request by Julien00859**
Changing the format of a data stored inside a cookie makes is dangerous
as there always is a possibility that someone has a valid session with
that cookie but with the previous format. Ideally in all places where we
use that cookie, we should test for all formats.

In this work we propose a more definitive solution to this problem.
Whenever one is going to use `request.cookies` we run code to sanitize
the existing cookies to converge all possible data format to a single
one.

This way the applications can not bother about what format their data is
and always process a single unified format.

### CODE REVIEW

File: odoo/http.py [hunk 1]
```diff
@@ -1557,6 +1557,13 @@ def best_lang(self):
         except (ValueError, KeyError):
             return None
 
+    @lazy_property
+    def cookies(self):
+        cookies = werkzeug.datastructures.MultiDict(self.httprequest.cookies)
+        if self.env:
+            self.env['ir.http']._sanitize_cookies(cookies)
```
Comment (Julien00859) [inline odoo/http.py:1564]: Maybe we should do
```py
if self.registry:
    self.registry['ir.http.']._sanitize_cookies
```

**Comment by jpp-odoo** [comment]
@robodoo rebase-ff

**Review by Julien00859 — COMMENTED** [review]
@robodoo r+

**Comment by Julien00859** [comment]
@adwid

**Pull request closed without merging**